### DISCUSSION TITLE
Is there a way to assign & revoke scripts to multiple users?

### DISCUSSION THREAD

**Discussion by SergioHLX** [violentmonkey/violentmonkey#1846] [General]
Is there a way to assign & revoke scripts to multiple users? As far as I can tell, all of the sync options are bidirectional.  So if I have multiple users synced to the same folder, then all of them can see all of the scripts & all of them can modify the scripts with those changes being pushed out to other users.

Alternatively, I've started Installing from URL.  This, means that different users can have different scripts & also that their changes don't get pushed back out.  But it leaves me with the responsibility of going computer-to-computer to install new scripts or revoke access.  When the user has decided to install on their personal computer, I can't do that. 

Is there a mechanism to assign & revoke scripts globally?  I thought that the WebDAV server would do it.  At least there I can assign different usernames & assign permissions on the server-side. But that also appears to be bidirectional.  

Is it possible to globally orchestrate this or am I stuck doing it manually?

**Comment by gera2ld** [comment]
Do you mean to always sync from the server to the client side and ignore local changes?

**Reply by SergioHLX** [reply]
Exactly, yes.  My goal is to only ask the user to log into a sync service (webdav/dropbox/etc) from Violentmonkey but not require them to go through the process of installing scripts. 

Though as I attempted to work on it, I realized there is more than one problem.  If I create a script on Dropbox (for instance) it is not pushed to the user.  The script needs to be in a different format created by the extension. I'm not even sure that it is possible to assign scripts. 

The use-case that I am working on is where there is one administrator with many scripts as well as many users in a company.  However

- Not all users should have all scripts
- Scripts must be routinely revoked as part of the sync without requiring the user to delete it.
- Most users are not technical & would not know how to install, update & manage their list of scripts.

**Reply by gera2ld** [reply]
Unfortunately it's not supported for now. But unidirectional syncing is a planned feature.
